bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaAccountHandler.java

```diff
@@ -0,0 +1,550 @@
+                            .post(Entity.entity(payLoad, MediaType.APPLICATION_JSON_TYPE));
+                }
+            } else {
+                if (command != null) {
+                    response = target.resolveTemplate("cmd", command).resolveTemplate("vid", vehicleId)
+                            .request(MediaType.APPLICATION_JSON_TYPE)
+                            .header("Authorization", "Bearer " + logonToken.access_token).get();
+                } else {
+                    response = target.resolveTemplate("vid", vehicleId).request(MediaType.APPLICATION_JSON_TYPE)
+                            .header("Authorization", "Bearer " + logonToken.access_token).get();
+                }
+            }
+
+            JsonParser parser = new JsonParser();
+
+            if (!checkResponse(response, false)) {
+                logger.debug("An error occurred while communicating with the vehicle during request {}: {}:{}",
+                        new Object[] { command, (response != null) ? response.getStatus() : "",
+                                (response != null) ? response.getStatusInfo() : "No Response" });
+                return null;
+            }
+
+            try {
+                JsonObject jsonObject = parser.parse(response.readEntity(String.class)).getAsJsonObject();
+                logger.trace("Request : {}:{}:{} yields {}",
+                        new Object[] { command, payLoad, target.toString(), jsonObject.get("response").toString() });
+                return jsonObject.get("response").toString();
+            } catch (Exception e) {
+                logger.error("An exception occurred while invoking a REST request : '{}'", e.getMessage());
+            }
+        }
+
+        return null;
+    }
+
+    protected Runnable connectRunnable = () -> {
+        try {
+            lock.lock();
+
+            if (getThing().getStatus() != ThingStatus.ONLINE) {
+                logger.debug("Setting up an authenticated connection to the Tesla back-end");
+
+                ThingStatusInfo authenticationResult = authenticate();
+                updateStatus(authenticationResult.getStatus(), authenticationResult.getStatusDetail(),
+                        authenticationResult.getDescription());
+
+                if (authenticationResult.getStatus() == ThingStatus.ONLINE) {
+                    // get a list of vehicles
+                    Response response = vehiclesTarget.request(MediaType.APPLICATION_JSON_TYPE)
+                            .header("Authorization", "Bearer " + logonToken.access_token).get();
+
+                    if (response != null && response.getStatus() == 200 && response.hasEntity()) {
+                        updateStatus(ThingStatus.ONLINE);
+                        for (Vehicle vehicle : queryVehicles()) {
+                            Bridge bridge = getBridge();
+                            if (bridge != null) {
+                                List<@NonNull Thing> things = bridge.getThings();
```

Comment:
Please do not use NonNull but use NonNullByDefault as [described in the coding guidelines](https://www.openhab.org/docs/developer/development/guidelines.html#a-code-style)